plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportStateChangedObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private val registeredViewportStates = CopyOnWriteArrayList<ViewportState>()
```

Comment:
Maybe a small nit but if I see no reasons in using thread-safe struct here, it may even hide some untrivial bugs. Perhaps use regular array list here?

Reply:
Could you explain what untrivial bug would it be? we are allowing adding, removing and iterating the states, maybe or maybe not in the same thread, I think we should use a thread-safe struct here.